src/mono/mono/component/event_pipe-stub.c

```diff
@@ -6,6 +6,35 @@
 #include "mono/component/event_pipe.h"
 #include "mono/metadata/components.h"
 
+
+#ifdef HOST_WASM
+#include <emscripten.h>
+
+G_BEGIN_DECLS
+
+EMSCRIPTEN_KEEPALIVE gboolean
+mono_wasm_event_pipe_enable (const char *output_path,
+			     uint32_t circular_buffer_size_in_mb,
+			     const char *providers,
+			     /* EventPipeSessionType session_type = EP_SESSION_TYPE_FILE, */
+			     /* EventPipieSerializationFormat format = EP_SERIALIZATION_FORMAT_NETTRACE_V4, */
+			     /* bool */ gboolean rundown_requested,
+			     /* IpcStream stream = NULL, */
+			     /* EventPipeSessionSycnhronousCallback sync_callback = NULL, */
+			     /* void *callback_additional_data, */
+			     int32_t *out_session_id);
```

Comment:
Only support for 32-bit session id's? EnvetPipe uses 64-bit id's and it is mapped to/from a pointer to underlying session struct.

Reply:
did I get it wrong? I thought the session ID was just a pointer cast to an int.  (wasm is 32-bit).

but yea we can use a wider session ID

Reply:
Under the hood its a pointer casted to an EventPipeSessionID that is typed to uint64_t on all platforms.